### PR TITLE
Feature/dropdown

### PR DESCRIPTION
**Pull request by AndreiHorhogea**
Created dropdown

### CODE REVIEW

File: src/components/ScoreByGameId.js [hunk 1]
```diff
@@ -1,42 +1,59 @@
 import React from "react";
 import Requests from "./Requests";
-import "../styles/scores.css"
+import "../styles/scores.css";
+
+// import Scores from "../components/Scores";
```
Comment (vlionescu) [inline src/components/ScoreByGameId.js]: delete commented code

File: src/components/ScoreByGameId.js [hunk 2]
```diff
@@ -1,42 +1,59 @@
...
+        );
+    }
 }
 
-export default ScoresByGameId;
\ No newline at end of file
+export default ScoresByGameId;
+
+
+
```
Comment (vlionescu) [inline src/components/ScoreByGameId.js]: delete empty lines

File: src/components/Scores.js [hunk 3]
```diff
@@ -17,16 +17,19 @@ class Scores extends React.Component {
...
   getScoresInList = () => {
     const list = this.state.games.map((game, index) => {
       return (
-        <option key={index} value={game.id}>
+        <option key={index} value={game.id}
+        >
           {game.name}
         </option>
+
+
```
Comment (vlionescu) [inline src/components/Scores.js]: delete empty rows

File: src/components/Scores.js [hunk 4]
```diff
@@ -37,13 +40,14 @@ class Scores extends React.Component {
     return (
       <select
         className="select-css"
-        value={this.state.selectedvalue}
+        value={this.state.selectedValue}
         onChange={e => this.onChangeHandle(e.target.value)}
-      >
+       >
+        <option value=''>Score</option>
         >{this.getScoresInList()}
```
Comment (vlionescu) [inline src/components/Scores.js]: there seems to be an extra ">" here

File: src/styles/dropdown.css [hunk 5]
```diff
@@ -10,7 +10,7 @@
     max-width: 100%;
     box-sizing: border-box;
     margin: 0;
-    border: 1px solid #aaa;
+    /*border: 1px solid #aaa;*/
```
Comment (vlionescu) [inline src/styles/dropdown.css]: delete commented code

File: src/styles/scores.css [hunk 6]
```diff
@@ -1,4 +1,16 @@
 .scoreListItem{
     color:white;
-    font-size: 30px;
-}
\ No newline at end of file
+    font-size:30px;
+    /*margin: normal20px;*/
+    /*margin-left: 500px;*/
+    text-align: center;
```
Comment (vlionescu) [inline src/styles/scores.css]: same here

File: src/styles/scores.css [hunk 7]
```diff
@@ -1,4 +1,16 @@
...
+}
+
+.scoreListClass{}
+
+.gameListClass{}
+
+/*=======*/
+/*    font-size: 30px;*/
+/*}*/
```
Comment (vlionescu) [inline src/styles/scores.css]: and here

File: src/components/ScoreByGameId.js [hunk 8]
```diff
@@ -1,42 +1,59 @@
...
+        this.state = {
+            id: '',
+            scores: []
+        };
+    }
+
+
+    async componentDidMount() {
+        const id = this.props.match.params.id;
+        this.setState({id, scores: await Requests.get("/scores/" + id)});
```
Comment (bogdandan) [inline src/components/ScoreByGameId.js]: Please handle erros

File: src/components/ScoreByGameId.js [hunk 9]
```diff
@@ -1,42 +1,59 @@
...
+    }
+
+    async componentDidUpdate(nextProps, prevState) {
```
Comment (bogdandan) [inline src/components/ScoreByGameId.js]: Is componentDidUpdate necesarly ?

File: src/components/ScoreByGameId.js [hunk 10]
```diff
@@ -1,42 +1,59 @@
...
+        const id = this.props.match.params.id;
+        if (id !== prevState.id) {
+            this.setState({id, scores: await Requests.get("/scores/" + id)});
```
Comment (bogdandan) [inline src/components/ScoreByGameId.js]: Please add the logic extract the setState to a method that's called from componentDidMount and componentDidUpdate

File: src/components/ScoreByGameId.js (hunk 9 above)
Comment (AndreiHorhogea) [reply src/components/ScoreByGameId.js]: yes

Comment (vlionescu) [reply src/components/ScoreByGameId.js]: The idea is we use this as a trigger when the url has changed to get the new scores, if there is a better solution for this please share.

**Pull request merged**